src/plugins/dashboard_embeddable_container/public/url_generator.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TimeRange, Filter, Query } from '../../data/public';
+import { setStateToKbnUrl } from '../../kibana_utils/public';
+import { UrlGeneratorsDefinition } from '../../share/public';
+
+export const STATE_STORAGE_KEY = '_a';
+export const GLOBAL_STATE_STORAGE_KEY = '_g';
+
+export const DASHBOARD_APP_URL_GENERATOR = 'DASHBOARD_APP_URL_GENERATOR';
+
+export interface DashboardAppLinkGeneratorState {
+  State: {
+    dashboardId?: string;
+    timeRange?: TimeRange;
+    filters?: Filter[];
+    query?: Query;
+    // If not given, will use the uiSettings configuration
+    useHash?: boolean;
+  };
+}
+
+export const createDirectAccessDashboardLinkGenerator = (
+  getStartServices: () => Promise<{ appBasePath: string; useHashedUrl: boolean }>
+): UrlGeneratorsDefinition<typeof DASHBOARD_APP_URL_GENERATOR> => ({
+  id: DASHBOARD_APP_URL_GENERATOR,
+  createUrl: async state => {
+    const startServices = await getStartServices();
+    const useHash = state.useHash || startServices.useHashedUrl;
+    const appBasePath = startServices.appBasePath;
+    const hash = state.dashboardId ? `dashboard/${state.dashboardId}` : `dashboard`;
+
+    const appStateUrl = setStateToKbnUrl(
+      STATE_STORAGE_KEY,
+      {
+        query: state.query,
+        filters: state.filters,
+      },
+      { useHash },
+      `${appBasePath}#/${hash}`
+    );
+
+    return setStateToKbnUrl(
+      GLOBAL_STATE_STORAGE_KEY,
+      {
+        timeRange: state.timeRange,
```

Comment:
Is this a change in the name of the global state parameter which stores the `to` and `from` times for the target dashboard? Testing the URL generated for ML, the URL now uses `timeRange`:

```
kibana#/dashboard/2ce94e70-58af-11ea-b552-c7abe8a1122c?_a=(filters:!(),query:(language:kuery,query:'airline:"$airline$"'))&_g=(timeRange:(from:'$earliest$',mode:absolute,to:'$latest$'))
```

where before it used `time`:

```
kibana#/dashboard/2ce94e70-58af-11ea-b552-c7abe8a1122c?_g=(time:(from:'$earliest$',mode:absolute,to:'$latest$'))&_a=(filters:!(),query:(language:kuery,query:'airline:"$airline$"'))
```

The result seems to be that the range is not set correctly on the target dashboard, and it opens with the default of Last 15 minutes. The URL that is getting generated is e.g.

```
http://localhost:5601/ioq/app/kibana#/dashboard/2ce94e70-58af-11ea-b552-c7abe8a1122c?_a=(filters:!(),query:(language:kuery,query:'airline:%22AAL%22'))&_g=(timeRange:(from:'2019-02-09T14:15:00.000Z',mode:absolute,to:'2019-02-09T18:30:00.000Z'))
```

Reply:
ah thank you, great catch, it's a mistake.  Problem with jest snapshot tests - the snapshot was wrong.